calendar/page.js

```diff
@@ -472,6 +505,10 @@ function thirtyMinutesFrom(date) {
   return new Date(date.getTime() + 30 * 60 * 1000);
 }
 
+function hasNonzeroTime(date) {
```

Comment:
There timezones that are shifted by minutes, so this won't work in all cases, but I think this is good enough for now.